### PR TITLE
Everywhere: Switch to clang-format-18 as the standard formatter

### PR DESCRIPTION
**Pull request by trflynn89**

### CODE REVIEW

File: AK/StdLibExtraDetails.h [hunk 1]
```diff
@@ -126,9 +126,9 @@ inline constexpr bool IsFunction<Ret(Args...) const volatile&> = true;
 template<class Ret, class... Args>
 inline constexpr bool IsFunction<Ret(Args..., ...) const volatile&> = true;
 template<class Ret, class... Args>
-inline constexpr bool IsFunction<Ret(Args...)&&> = true;
+inline constexpr bool IsFunction<Ret(Args...) &&> = true;
```
Comment (Hendiadyoin1) [inline AK/StdLibExtraDetails.h:129]: This change feels weird

File: Kernel/Bus/VirtIO/Device.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ UNMAP_AFTER_INIT ErrorOr<void> Device::initialize_virtio_resources()
 }
 
 UNMAP_AFTER_INIT VirtIO::Device::Device(NonnullOwnPtr<TransportEntity> transport_entity)
-    : m_class_name(transport_entity->determine_device_class_name())
+    : m_class_name(transport_entity -> determine_device_class_name())
```
Comment (Hendiadyoin1) [inline Kernel/Bus/VirtIO/Device.cpp:30]: Same here

File: Kernel/Devices/GPU/Console/VGATextModeConsole.cpp [hunk 3]
```diff
@@ -20,7 +20,7 @@ UNMAP_AFTER_INIT NonnullLockRefPtr<VGATextModeConsole> VGATextModeConsole::initi
 UNMAP_AFTER_INIT VGATextModeConsole::VGATextModeConsole(NonnullOwnPtr<Memory::Region> vga_window_region)
     : Console(80, 25)
     , m_vga_window_region(move(vga_window_region))
-    , m_current_vga_window(m_vga_window_region->vaddr().offset(0x18000).as_ptr())
+    , m_current_vga_window(m_vga_window_region -> vaddr().offset(0x18000).as_ptr())
```
Comment (Hendiadyoin1) [inline Kernel/Devices/GPU/Console/VGATextModeConsole.cpp:23]: .

File: Tests/JSSpecCompiler/test-runner.cpp [hunk 4]
```diff
@@ -59,11 +59,11 @@ const Array regression_tests = {
     }
 };
 
-static const LexicalPath path_to_compiler_binary = [] {
+static LexicalPath const path_to_compiler_binary = [] {
```
Comment (Hendiadyoin1) [inline Tests/JSSpecCompiler/test-runner.cpp:62]: huh I see why east const applies here, but it feels somewhat weird with the `static`

File: Userland/Libraries/LibWasm/Types.h [hunk 5]
```diff
@@ -495,9 +494,8 @@ class Instruction {
         i32,
         i64,
         u128,
-        u8 // Empty state
-    > m_arguments;
-    // clang-format on
+        u8> // Empty state
+        m_arguments;
```
Comment (Hendiadyoin1) [inline Userland/Libraries/LibWasm/Types.h:498]: This makes me wonder why the empty state is a `u8`
@alimpfard

**Review by Hendiadyoin1 — COMMENTED** [review]
Also new east volatile?

File: Kernel/Bus/VirtIO/Device.cpp (hunk 2 above)
Comment (trflynn89) [reply Kernel/Bus/VirtIO/Device.cpp:30]: I pointed this out in the commit message. It is weird, but not worth the ugliness of a `clang-format off` the couple of times it appears.

File: Tests/JSSpecCompiler/test-runner.cpp (hunk 4 above)
Comment (trflynn89) [reply Tests/JSSpecCompiler/test-runner.cpp:62]: east-const-ness has nothing to do with static-ness

File: AK/StdLibExtraDetails.h (hunk 1 above)
Comment (trflynn89) [reply AK/StdLibExtraDetails.h:129]: The `&&` here is a ref qualifier on the function type, the extra space is actually the right thing to do. It's akin to functions like:
```c++
[[nodiscard]] StringView bytes_as_string_view() const&;
[[nodiscard]] StringView bytes_as_string_view() const&& = delete;
```

File: Kernel/Bus/VirtIO/Device.cpp (hunk 2 above)
Comment (alimpfard) [inline Kernel/Bus/VirtIO/Device.cpp:30]: 🤔 there's exactly one instance of this I can see, maybe it'd go away if we put the thing in another set of parens?
not a blocker obv, just thinking out loud.

File: Userland/Libraries/LibGfx/Font/OpenType/Tables.cpp [hunk 6]
```diff
@@ -426,7 +426,8 @@ bool OS2::use_typographic_metrics() const
 Optional<i16> OS2::x_height() const
 {
     return m_data.visit(
-        []<typename T> requires(requires { T::sx_height; })(T * data)->Optional<i16> {
+        []<typename T>
+        requires(requires { T::sx_height; })(T * data) -> Optional<i16> {
```
Comment (alimpfard) [inline Userland/Libraries/LibGfx/Font/OpenType/Tables.cpp:430]: Eyyy, some actually good improvement!

**Review by alimpfard — APPROVED** [review]

File: Kernel/Bus/VirtIO/Device.cpp (hunk 2 above)
Comment (trflynn89) [reply Kernel/Bus/VirtIO/Device.cpp:30]: There's another in Kernel/Devices/GPU/Console/VGATextModeConsole.cpp
Parens don't fix it unfortunately.

Comment (alimpfard) [reply Kernel/Bus/VirtIO/Device.cpp:30]: Huh. alrighty then.

File: Userland/Services/LookupServer/MulticastDNS.h [hunk 7]
```diff
@@ -38,7 +38,6 @@ class MulticastDNS : public Core::UDPServer {
 
     // https://github.com/llvm/llvm-project/issues/56685
     // https://github.com/llvm/llvm-project/pull/65409
```
Comment (ADKaster) [inline Userland/Services/LookupServer/MulticastDNS.h:40]: Are these two issues clang-format ones?

**Review by ADKaster — COMMENTED** [review]
One comment question, otherwise lgtm

File: Userland/Services/LookupServer/MulticastDNS.h (hunk 7 above)
Comment (trflynn89) [reply Userland/Services/LookupServer/MulticastDNS.h]: Ah yep! removed

**Pull request merged**